src/plugins/template/backend/ops/convolution.cpp

```diff
@@ -14,11 +15,18 @@ bool evaluate(const std::shared_ptr<ov::op::v1::Convolution>& op,
               const ov::TensorVector& inputs) {
     using ET = typename ov::element_type_traits<T>::value_type;
     const auto filter_data = inputs[1].data<ET>();
-    auto out_data_ptr = outputs[0].data<ET>();
     const auto in_data_ptr = inputs[0].data<ET>();
-    const auto& out_shape = outputs[0].get_shape();
     const auto& in_shape = inputs[0].get_shape();
     const auto& filter_shape = inputs[1].get_shape();
+
+    const auto input_shapes = ov::util::get_tensors_partial_shapes(inputs);
+    auto bpads_cp = ov::CoordinateDiff();
+    auto endpads_cp = ov::CoordinateDiff();
+    const auto output_shape = ov::op::v1::shape_infer(op.get(), input_shapes, bpads_cp, endpads_cp).front();
```

Comment:
The previous idea to have generic shape inference (if required) in template plugin was good but maybe is required to experiment with it as separate PR.